save-highscores.s

```diff
@@ -0,0 +1,60 @@
+.include "build/tetris.inc"
+.include "ips.inc"
+
+SRAM := $6000
+
+; patch to load from SRAM (and use a length instead of a terminator)
+
+.segment "initHighScoreTable_patch"
+        ips_segment     "initHighScoreTable_patch",$8095
+
+; initHighScoreTable_patch:
+initHighScoreTable:
+        lda SRAM,x
+        cpx #$50
+        beq continueColdBootInit
+        sta highScoreNames,x
+        inx
+        jmp initHighScoreTable
+
+continueColdBootInit:
+
+; start button patch to add saving
+
+.segment "saveHighscoreStartPressed_patch"
+        ips_segment     "saveHighscoreStartPressed_patch",$A337
+
+; saveHighscoreStartPressed_patch:
+        jsr saveHighScores
+
+
+; routine for storing the high score data
+
+.segment "saveHighScores"
+        ips_segment     "saveHighScores",unreferenced_data4
+
+saveHighScores:
+        ldx #$00
+@loop:
+        lda highScoreNames,x
+        sta SRAM,x
+        inx
+        cpx #$51
+        bne @loop
+        jsr updateAudioWaitForNmiAndResetOamStaging
+        rts
+
+; header patch to enable SRAM
+
+.fileopt    comment, "ips: 0 INES_HEADER_MOD"
+.segment "INES_HEADER_MOD_HDR"
+.byte 0
```

Comment:
nit: Normally I'd indent these (.byte and .dbyt). Although there's probably plenty of places I've forgotten to.